src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
+	}
+
+	public void resetBasket() {
+		buyBasket();
+		codeDePromotion = false;
+	}
+
+	public void buyBasket() {
+		map.clear();
+	}
+
+	public boolean isBasketContains(String produit) {
+		boolean found = false;
+		for (String s : map.keySet()) {
+			if (s == produit) found = true;
+		}
+		return found;
+	}
+
+	public void mixWithBasket(BasketInformations b) {
+		map.putAll(b.map);
+	}
+
+	public boolean isSendFree() {
+		boolean free = false;
+
+		//Contains 3 or more local products
+		if(totalLocalProducts(0,0) > 3) {
+			free = true;
+		}
+
+		//Product exceeds 50
+		int max = 0;
+		for(String s : map.keySet()){
+			if(map.get(s) > max){
+				max = map.get(s);
+			}
+		}
+		if(max > 50){
+			free = true;
+		}
```

Comment:
```suggestion
		if(map.values().stream().anyMatch(value -> value > 50.0)){
			free = true;
		}
```


To find if an element that corresponds to certain criteria exists inside a Map we can achieve like this:

map.values().stream().anyMatch(value -> value > 50.0);

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Stream.html#anyMatch(java.util.function.Predicate)